src/util.rs

```diff
@@ -65,3 +65,16 @@ pub fn read_tag<R: BufRead + Seek>(reader: &mut R) -> ImageResult<(String, usize
 
     Ok((String::from_utf8_lossy(&tag_buf).into_owned(), size))
 }
+
+pub fn read_null_terminated_string<R: Read>(reader: &mut R) -> io::Result<String> {
```

Comment:
I don't know if it's possible to make this `BufRead` instead, but if so you can use [read_until](https://doc.rust-lang.org/std/io/trait.BufRead.html#method.read_until).